### ISSUE TITLE
`kubectl patch` shows "not patched" even after the resource is patched successfully

### ISSUE THREAD

**Issue by janetkuo** [kubernetes/kubernetes#42494]
```console
$ kubectl version 
Client Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.0-alpha.0.781+a2c7eb275443a5-dirty", GitCommit:"a2c7eb275443a5afea4052094a07daf627e4a724", GitTreeState:"dirty", BuildDate:"2017-03-03T19:01:20Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7+", GitVersion:"v1.7.0-alpha.0.781+a2c7eb275443a5-dirty", GitCommit:"a2c7eb275443a5afea4052094a07daf627e4a724", GitTreeState:"dirty", BuildDate:"2017-03-03T19:01:20Z", GoVersion:"go1.7.3", Compiler:"gc", Platform:"linux/amd64"}

$ kubectl run nginx --image nginx 
deployment "nginx" created
// progressDeadlineSeconds was 600 by default 

$ kubectl patch deployment nginx -p '{"spec":{"progressDeadlineSeconds":5}}'
deployment "nginx" not patched
// progressDeadlineSeconds is changed to 5 successfully
```

Better fix this in 1.6. @kubernetes/sig-cli-bugs

**Comment by ymqytw** [comment]
Dup of #42384

**Comment by ymqytw** [comment]
The fix #42391 is in the merge queue

**Comment by janetkuo** [comment]
Cool, closing in favor of #42384
